frontend/src/concepts/dashboard/DashboardModalFooter.tsx

```diff
@@ -15,9 +16,19 @@ type DashboardModalFooterProps = {
   isSubmitDisabled: boolean;
   isCancelDisabled?: boolean;
   alertTitle: string;
-  error?: Error;
+  error?: Error | AxiosError;
 };
 
+const isAxiosErrorWithInnerMessage = (
```

Comment:
Can we turn this into a parse layer at the axios usage -- I'd like to drop axios and I don't think the world of components needs to know about axios' internal usage. Just throw a new error in the `.catch` with a parsed value.

Reply:
I'd imagine you could even make a utility `throwErrorFromAxios` which parses the error out of the `.catch` and then throws it from the utility -- which should still bubble back up.

Something like:
```
axios(...)
  .then(() => /* some then functionality */)
  .catch(throwErrorFromAxios)
```

Reply:
Hmm, yeah this does seem like a better separation of concerns. It won't have the benefit of improving the other errors outside the scope of these namespace updates, but maybe that's a good thing. I'm also hesitant to make that change for all axios usages, at least in this PR. It might be worth trying that in a standalone PR where we can easily revert it if it causes trouble elsewhere.

Reply:
That's fair -- don't worry about the wide axios usage -- let us just go with this particular call and log a Task issue for tech debt cleanup.

Ideally I'd like to drop axios in the long run, as it predates my chosen path and eventually I'd like to lean 100% into proxy utils and the SDK utils... prometheus calls also are narrow now that we have UWM stuff setup from the stuff Alex did. Axios to call our node instance for custom business logic should go away in https://issues.redhat.com/browse/RHOAIENG-1461

Reply:
@andrewballantyne I think maybe for this `throwErrorFromAxios` utility we may not want to entirely discard the outer AxiosError object and throw only the inner error response, because I'm not sure what other code might be expecting a true `AxiosError` with all its properties. What about instead having the utility just look for an inner `message` if present and rewrite the `message` at the top, retaining the rest of the `AxiosError`? I'll play with that and push a change to show you what I mean.

Reply:
Pushed. I added the following to `frontend/src/api/errorUtils.ts`: 

```ts
const isAxiosErrorWithResponseMessage = (
  error?: Error | AxiosError,
): error is AxiosError<{ message: string }> =>
  Boolean(
    error && typeof (error as AxiosError<{ message: string }>).response?.data?.message === 'string',
  );

export const throwErrorFromAxios = (error?: Error | AxiosError) => {
  if (isAxiosErrorWithResponseMessage(error)) {
    throw { ...error, message: error.response?.data.message };
  }
  throw error;
};
```

and then used it in the `catch` as you described above, but only for this one request.

Reply:
Hmm, it looks like we only actually call `axios()` directly in one other place: when creating a data science project. I may as well use this there too.

Reply:
Could we throw `new Error` like objects? And try to not throw the Axios error unless it has a `.message` on it

Reply:
I wasn't sure if we wanted to clobber anything else on the existing error object that we might want in the components, but I suppose we're only really reading the `.message` anyway. I'll change it so it throws a `new Error` with the response data object only, if present, and otherwise falls back to throwing the AxiosError